group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -452,21 +453,38 @@ public Group group(String groupId, long committedOffset) throws GroupIdNotFoundE
     /**
      * Get the Group List.
      *
-     * @param statesFilter The states of the groups we want to list.
-     *                     If empty all groups are returned with their state.
-     * @param committedOffset A specified committed offset corresponding to this shard
+     * @param statesFilter      The states of the groups we want to list.
+     *                          If empty, all groups are returned with their state.
+     * @param typesFilter       The types of the groups we want to list.
+     *                          If empty, all groups are returned with their type.
+     * @param committedOffset   A specified committed offset corresponding to this shard.
      *
      * @return A list containing the ListGroupsResponseData.ListedGroup
      */
+    public List<ListGroupsResponseData.ListedGroup> listGroups(
+        Set<String> statesFilter,
+        Set<String> typesFilter,
+        long committedOffset
+    ) {
+        // Converts each string to a value in the GroupType enum while being case-insensitive.
+        Set<Group.GroupType> enumTypesFilter = typesFilter.stream()
+            .map(Group.GroupType::parse)
+            .collect(Collectors.toSet());
```

Comment:
Yeah I was actually unsure about this as well. 
In the consumerGroupCommand tool we return an illegalArgumentException if the type is unknown so I thought maybe it would make sense to do that here as well. To ignore it should parse return null? If it returns null I was worried about NullPointerExceptions. If I wanted to return UNKNOWN type, we have to add that to the enum on the server side. Is there some obvious way to do this that I'm missing?

Reply:
We were able to ignore the unknown states easily because we didn't have to parse them and assign an enum type right? It would just check if the state exists in the filter set, but here since we're tryna parse and assign ig it needs to be handled differently? Lmk if my understanding is off

Reply:
> In the consumerGroupCommand tool we return an illegalArgumentException if the type is unknown so I thought maybe it would make sense to do that here as well.

Yeah, the context is a bit different here. You have to think about how illegalArgumentException will get returned to the user. The Kafka protocol does not have any error for this so it converts it to a Unknown Server Error without any details. This is not great from a user perspective.

I think that the correct way would be to try parsing the received group type. If it succeeds, great, we keep it. Otherwise, we simply discard it. If you want to keep the illegalArgumentException, you could catch it when you call `parse`.

Reply:
I initially threw the illegalArgException and then caught it and ignored it in the listGroups method but this caused the opposite result from what we wanted aka it returns all the groups since the typeFilter is empty. The only way to get the expected behavior is to add unknown to the group type enum. This gives us the same result as when an unknown state filter is passed. I even added tests to make sure. Lmk if this doesn't work or if there was something else you had in mind.